test/soca/CMakeLists.txt

```diff
@@ -11,3 +11,79 @@ ecbuild_add_test(
       ENVIRONMENT
         OBS_DIR=${OBS_DIR}
 )
+
+# Symlink test input yaml files
+# -----------------------------
+# create testinput dir
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testinput)
+
+# list of test input files to install
+list(APPEND test_input
+  ${PROJECT_SOURCE_DIR}/test/soca/testinput/exp_obs.yaml
+  ${PROJECT_SOURCE_DIR}/test/soca/testinput/dumy.yaml
+)
+
+# symlink
+foreach(FILENAME ${test_input})
+  file(RELATIVE_PATH FILENAME_REL ${PROJECT_SOURCE_DIR}/test/soca/testinput/ ${FILENAME})
+  file(CREATE_LINK ${FILENAME} ${PROJECT_BINARY_DIR}/test/soca/testinput/${FILENAME_REL} SYMBOLIC)
+endforeach(FILENAME)
+
+# install
+install(FILES ${test_input}
+        DESTINATION "test/testinput/")
+
+# Set runtime environement variables
+# ----------------------------------
+set( CDATE 2018041512 )        # Analysis time
+set( GDATE 2018041509 )        # What is that?
+set( assim_freq 6 )            # DA window
+set( OBS_LIST "adt_j3 sst_noaa19_l3u icec_EMC icefb_GDR temp_profile_fnmoc salt_profile_fnmoc" )
+set( COMOUT ${CMAKE_CURRENT_BINARY_DIR}/3dvar )
+set( COMIN_OBS ${CMAKE_CURRENT_BINARY_DIR}/obs/r2d2-shared )
+set( COMIN_GES ${CMAKE_CURRENT_BINARY_DIR}/bkg )  # TODO: stage the background somewhere
+set( CASE "12345" )                               # TODO: No idea what this is, figure it out!
+set( CASE_ENKF "123456")                          # TODO: No idea what this is, figure it out!
+set( LEVS "75" )                                  # TODO: I can sort of guess ...
+set( OBS_YAML ${CMAKE_CURRENT_BINARY_DIR}/testinput/exp_obs.yaml ) # TODO: Is that the list of obs
```

Comment:
This is a list of YAML files that contain all the obs, qc, etc. stuff that will be concatenated together. The old prep job would use the EWOK yamls to get the `obsdatain` path, but we could agree to some convention and just make it a list of ob types to simplify things